source/fuzz/fuzzer_pass_add_dead_breaks.cpp

```diff
@@ -34,24 +34,39 @@ void FuzzerPassAddDeadBreaks::Apply() {
   // We consider each function separately.
   for (auto& function : *GetIRContext()->module()) {
     // For a given function, we find all the merge blocks in that function.
-    std::vector<uint32_t> merge_block_ids;
+    std::vector<opt::BasicBlock*> merge_blocks;
     for (auto& block : function) {
       auto maybe_merge_id = block.MergeBlockIdIfAny();
       if (maybe_merge_id) {
-        merge_block_ids.push_back(maybe_merge_id);
+        auto merge_block =
+            fuzzerutil::MaybeFindBlock(GetIRContext(), maybe_merge_id);
+        if (merge_block == nullptr) {
+          continue;
+        }
+
+        merge_blocks.push_back(merge_block);
       }
     }
     // We rather aggressively consider the possibility of adding a break from
     // every block in the function to every merge block.  Many of these will be
     // inapplicable as they would be illegal.  That's OK - we later discard the
     // ones that turn out to be no good.
     for (auto& block : function) {
-      for (auto merge_block_id : merge_block_ids) {
-        // TODO(https://github.com/KhronosGroup/SPIRV-Tools/issues/2856): right
-        //  now we completely ignore OpPhi instructions at merge blocks.  This
-        //  will lead to interesting opportunities being missed.
+      for (auto* merge_block : merge_blocks) {
+        std::vector<uint32_t> phi_ids;
```

Comment:
I would prefix this with a comment saying that this will be populated with ids available at the branch point that can be used to update OpPhi instructions at the branch target.

Reply:
Fixed in a04b185.